src/main/java/ca/wolfram/beta/symath/VMap.java

```diff
@@ -0,0 +1,24 @@
+package ca.wolfram.beta.symath;
+import java.util.HashMap;
+
+/**
+ * Our own hashmap implementation to warn users when overwriting values
+ */
+public class VMap extends HashMap<String, Double> {
+    @Override
+    public Double put(String key, Double value) {
+        if (containsKey(key))
+            System.out.println(String.format("Caution: %s was already added with a value of %f", key, get(key)));
+        return super.put(key, value);
+    }
+
+    public double get(String key, double fallback) {
+        if (containsKey(key)) return get(key);
```

Comment:
if we do override get, odds are we'll want this get to behave the same way as our implementation